trustfall_core/src/interpreter/hints/vertex_info.rs

```diff
@@ -14,6 +14,17 @@ use crate::{
 
 use super::{dynamic::DynamicallyResolvedValue, CandidateValue, EdgeInfo, Range};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct RequiredProperty {
+    pub name: Arc<str>,
```

Comment:
Not sure if we should add more fields such as property type or not

Reply:
Do you foresee a situation where some optimization would want to know the type of the property as well as its name?

If you can't think of any right now, I think it's fine to not add property types right now but "leave room" for adding them later if needed. An example of leaving room is making the struct non-exhaustive so that such additions aren't major changes.

Reply:
Not from the top of my head, so "leaving room" for now